src/mixxx.cpp

```diff
@@ -758,6 +758,16 @@ void MixxxMainWindow::finalize() {
         leakedControls.clear();
     }
 
+    // Delete the track collections after all internal track pointers
+    // in other components have been released by deleting those components
+    // beforehand!
```

Comment:
Thank you for explicitly documenting those dependencies. That will probably save some trouble when someone does more refactoring of this file in the future.

Reply:
Getting the shutdown order right was one of the main drivers for decoupling `TrackCollectionManager` from `Library`.